ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -59,16 +72,40 @@
    * @throws IOException if there is an error connecting to the Helix cluster.
    */
   @Override
-  public void initialize(String hostName, int port) throws IOException {
+  public void initialize(String hostName, int port, List<HealthReport> healthReports) throws IOException {
     String instanceName = ClusterMapUtils.getInstanceName(hostName, port);
     manager = helixFactory.getZKHelixManager(clusterName, instanceName, InstanceType.PARTICIPANT, zkConnectStr);
     StateMachineEngine stateMachineEngine = manager.getStateMachineEngine();
     stateMachineEngine.registerStateModelFactory(LeaderStandbySMD.name, new AmbryStateModelFactory());
+    Map<String, TaskFactory> taskFactoryMap = new HashMap<>();
+    for (HealthReport healthReport : healthReports) {
+      if (healthReport.getAggregatePeriodInMinutes() != Utils.Infinite_Time) {
+        // enable cluster wide aggregation for the health report
+        final HelixClusterAggregator aggregator =
```

Comment:
Also move the task registration to a separate method. Let the initialize be simple for readability purposes. 

```
 stateMachineEngine.registerStateModelFactory(LeaderStandbySMD.name, new AmbryStateModelFactory());
 registerHealthReportTasks(healthReports);
    try {
      manager.connect();
    }
.
.
.
```